azurerm/internal/services/blueprints/tests/blueprint_assignment_resource_test.go

```diff
@@ -0,0 +1,395 @@
+func TestAccBlueprintAssignment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_blueprint_assignment", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMBlueprintAssignmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBlueprintAssignment_basic(data, "testAcc_basicSubscription", "v0.1_testAcc"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckBlueprintAssignmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccBlueprintAssignment_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_blueprint_assignment", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMBlueprintAssignmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBlueprintAssignment_basic(data, "testAcc_basicSubscription", "v0.1_testAcc"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckBlueprintAssignmentExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccBlueprintAssignment_requiresImport(data, "testAcc_basicSubscription", "v0.1_testAcc"),
+				ExpectError: acceptance.RequiresImportError("azurerm_blueprint_assignment"),
+			},
+		},
+	})
+}
+
+// Scenario: BP with RG's, locking and parameters/policies stored at Subscription, applied to subscription
+func TestAccBlueprintAssignment_subscriptionComplete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_blueprint_assignment", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMBlueprintAssignmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBlueprintAssignment_subscriptionComplete(data, "testAcc_subscriptionComplete", "v0.1_testAcc"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckBlueprintAssignmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+// Scenario: BP stored at Root Management Group, applied to Subscription
+func TestAccBlueprintAssignment_managementGroup(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_blueprint_assignment", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMBlueprintAssignmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBlueprintAssignment_rootManagementGroup(data, "testAcc_basicRootManagementGroup", "v0.1_testAcc"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckBlueprintAssignmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
```

Comment:
It'd be great to have a test that updates the resource as well just to confirm that all works